enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/RevisionHistoryEnricher.java

```diff
@@ -52,15 +48,14 @@ public RevisionHistoryEnricher(EnricherContext buildContext) {
 
     @Override
     public void addMissingResources(KubernetesListBuilder builder) {
-        final Integer maxRevisionHistories = Integer.parseInt(getConfig(Config.limit));
+        final Integer maxRevisionHistories = Configs.asInt(getConfig(Config.limit));
 
         log.info("Adding revision history limit to %s", maxRevisionHistories);
 
         builder.accept(new TypedVisitor<DeploymentBuilder>() {
             @Override
             public void visit(DeploymentBuilder item) {
-                DeploymentFluent.SpecNested<DeploymentBuilder> spec =
-                        item.getSpec() == null ? item.withNewSpec() : item.editSpec();
+                DeploymentFluent.SpecNested<DeploymentBuilder> spec = item.editOrNewSpec();
```

Comment:
Actually you could even elegantly nest it like in 

```java
item.editOrNewSpec()
      .withRevisionHistoryLimit(maxRevisionHistories)
    .endSpec();
```

so you don't need to declare the lengthy variable.

Reply:
fixed